service/collector/collector.go

```diff
@@ -6,12 +6,18 @@ import (
 	"sync"
 	"time"
 
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
 	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
 	"github.com/prometheus/client_golang/prometheus"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/controller/v4/key"
```

Comment:
@giantswarm/sig-monitoring here is some weirdness I mentioned in architecture review which is rather weird. Uncertain how to improve this. Once we run multiple operators via `chart-operator` and `release-operator` we are going to execute multiple collectors and the usage of the key package here is better, but the multiple collectors are going to overwrite themselves eventually. Questionmark. /cc @giantswarm/sig-releng